python/src/scipp/_shape.py

```diff
@@ -51,20 +51,76 @@ def concatenate(x, y, dim):
     return _call_cpp_func(_cpp.concatenate, x, y, dim)
 
 
-def reshape(x, dims, shape):
+def reshape(x, sizes=None, dims=None, shape=None):
     """Reshape a variable.
 
+    Examples:
+      a = sc.array(dims=['x'], values=[1, 2, 3, 4, 5, 6])
+      sc.reshape(var, sizes={'y': 2, 'z': 3})
+      sc.reshape(var, dims=['y', 'z'], shape=(2, 3))
+
     :param x: Variable to reshape.
-    :param dims: List of new dimensions.
-    :param shape: New extents in each dimension.
+    :param sizes: A dict mapping new dims to new shapes.
+    :param dims: A list of new dims labels.
+    :param shape: A list of new dim shapes.
     :type x: Variable
+    :type sizes: dict
     :type dims: list[str]
     :type shape: list[int]
     :raises: If the volume of the old shape is not equal to the
              volume of the new shape.
-    :return: New variable with requested dimension labels and shape.
+    :return: Variable with requested dimension labels and shape.
```

Comment:
Do we want to use "name" rather than "label" given this historic use of "label" in scipp?

Reply:
I think "dimension labels" is used throughout our docs.